src/components/text/text_color.tsx

```diff
@@ -38,7 +38,7 @@ export type EuiTextColorProps = CommonProps &
     /**
      * Any of our named colors or a `hex`, `rgb` or `rgba` value.
      */
-    color?: TextColor | CSSProperties['color'];
+    color?: TextColor | Exclude<CSSProperties['color'], 'inherit'>;
```

Comment:
@chandlerprall before this is what it was showing:

<img width="749" alt="Screenshot 2022-09-28 at 16 50 58" src="https://user-images.githubusercontent.com/2750668/192826413-c7d09f7b-d836-473b-b570-91081718dc27.png">


With the `Exclude<CSSProperties['color'], 'inherit'>;` now the `inherit` shows. But much more options... So not sure if there is something better than the `Exclude`



<img width="745" alt="Screenshot 2022-09-28 at 16 50 44" src="https://user-images.githubusercontent.com/2750668/192826439-3284c3ac-34fa-4c3a-8822-a27460f4461a.png">

Reply:
Can we try this instead to see how react docgen handles the output? 
```suggestion
    color?: TextColor | 'initial' | CSSProperties['color'];
```

Reply:
@constancecchen that's the output (`inherit` doesn't show): 

<img width="430" alt="Screenshot 2022-09-28 at 18 08 37" src="https://user-images.githubusercontent.com/2750668/192845825-a4c1ec6a-d350-4edb-8258-5c7936d32b75.png">

Reply:
@constancecchen The thought I was going to try out was if we could parse out `@type: A | B | C` from the description comment and override react docgen. Should work, but not positive it's a pattern we want to follow.

Reply:
😬 Yeah, I'm also a little leery of tinkering with docgen for this specific case. I leaving the options alone and using `@default` for `inherit` is clear enough, just IMO.

Reply:
Sounds good to me